saw-core/src/SAWCore/Term/CtxTerm.hs

```diff
@@ -889,26 +505,25 @@ ctxCtorElimType ret (a_top :: Proxy (Typ a)) (d_top :: DataIdent d) c
   -- a slightly richer type, but we are not going to try to compute this richer
   -- type in Haskell land.
   helper :: MonadTerm m =>
-    Proxy (Typ a) ->
-    DataIdent d ->
-    InvBindings CtxTerm EmptyCtx (ps ::> Arrows ixs (d -> Typ a)) ->
-    InvBindings CtxTerm (ps ::> Arrows ixs (d -> Typ a)) prevs ->
-    Bindings (CtorArg d ixs) ((ps ::> Arrows ixs (d -> Typ a)) <+> prevs) args ->
-    CtxTerms (CtxInvApp ((ps ::> Arrows ixs (d -> Typ a)) <+> prevs) args) ixs ->
-    m (CtxTerm ((ps ::> Arrows ixs (d -> Typ a)) <+> prevs) (Typ (Arrows args a)))
-  helper _a d params_pret prevs NoBind ret_ixs =
+    Name ->
+    [(LocalName, Term)] ->
+    [(LocalName, Term)] ->
+    [(LocalName, CtorArg)] ->
+    [Term] ->
+    m Term
+  helper d params_pret prevs [] ret_ixs =
     -- If we are finished with our arguments, construct the final result type
     -- (p_ret ret_ixs (c params prevs))
     do (vars, prev_vars) <- ctxVars2 params_pret prevs
        let (param_terms, p_ret) = ctxTermsCtxHeadTail vars
```

Comment:
How do we know that this call to `ctxTermsCtxHeadTail` is safe? Worth documenting how it upholds the precondition that `vars` is non-empty.

Reply:
Fixed in 94459a478425b5ea589e5817f60ef1473f4db7ef.